packages/react-sdk/src/props.ts

```diff
@@ -63,3 +65,49 @@ export const usePropAsset = (instanceId: Instance["id"], name: string) => {
   const asset = useStore(assetStore);
   return asset;
 };
+
+// this utility is used for link component in both builder and preview
+// so need to optimize rerenders with computed
+export const usePropUrl = (
+  instanceId: Instance["id"],
+  name: string
+): Page | string | undefined => {
+  const { propsByInstanceIdStore, pagesStore } = useContext(ReactSdkContext);
+  const assetStore = useMemo(() => {
+    return computed(
+      [propsByInstanceIdStore, pagesStore],
+      (propsByInstanceId, pages) => {
+        const instanceProps = propsByInstanceId.get(instanceId);
+        if (instanceProps === undefined) {
+          return undefined;
+        }
+        for (const prop of instanceProps) {
+          if (prop.name === name) {
+            if (prop.type === "page") {
+              return pages.get(prop.value);
+            }
+
+            if (prop.type === "string") {
+              for (const page of pages.values()) {
+                if (page.path === prop.value) {
+                  return page;
+                }
+              }
+              return prop.value;
+            }
+
+            return undefined;
+          }
+        }
+      }
+    );
+  }, [propsByInstanceIdStore, pagesStore, instanceId, name]);
+  const asset = useStore(assetStore);
+  return asset;
+};
+
+export const getInstanceIdFromComponentProps = (
+  props: Record<string, unknown>
+) => {
+  return props[idAttribute] as string;
```

Comment:
We pass `idAttribute` into all components. But we don't want it in components' props, because then it will show up in props panel

Reply:
hmm, if its our private instance id user shouldn't know about maybe we should mark it accordingly otherwise essentially components can't receive id user prop

Reply:
Not sure I understand. `idAttribute === "data-ws-id"`

Reply:
> components can't receive id user prop

So a component author can define a prop named `"id"`, no problem